addons/pos_stripe/static/src/js/payment_stripe.js

```diff
@@ -138,7 +153,11 @@ let PaymentStripe = PaymentInterface.extend({
                 return false;
             } else if (processPayment.paymentIntent) {
                 line.set_payment_status('waitingCapture');
-                await this.captureAfterPayment(processPayment, line);
+
+                if (! this._isInterac(processPayment, line)) {
+                    await this.captureAfterPayment(processPayment, line);
+                }
+
```

Comment:
Maybe it is more logic and easy  to make only one `if` ?

```
                let intentCharge = processPayment.paymentIntent.charges.data[0];

                if (intentCharge.payment_method_details.type === 'interac_present') {
                    this._isInterac(intentCharge, line)
                } else {
                    await this.captureAfterPayment(processPayment, line);
                }


    _isInterac: function (intentCharge, line) {
        // Canadian interac payments should not be captured:
        // https://stripe.com/docs/terminal/payments/regional?integration-country=CA#create-a-paymentintent

        line.card_type = 'interac';
        line.transaction_id = intentCharge.id;
    },
```

Or remove `_isInterac()` ?
use `let` instead `const` not really impact here

Reply:
Thanks for your review! It was indeed a bit weird that `_isInterac()` had side effects. I rewrote it to avoid that, while at the same time keeping `collectPayment()` simple by not digging too deep into the `processPayment` variable.